### PR TITLE
First pass at python logger

### PR DESCRIPTION
**Pull request by mschroering**
Started with what Simon built for patient-analytics and extended it to log the same JSON format that we use for our nodejs lambdas.

### CODE REVIEW

**Review by RyanRatcliff — DISMISSED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+language: python
+python:
+    - "3.7"
```
Comment (simons5593) [inline .travis.yml]: Just want to make sure you're aware variant-analytics is on python3.6. Any concern here if this is on 3.7?

Comment (mschroering) [reply .travis.yml]: I'll update to 3.6

File: lifeomic_logging/logger.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    ----------
+    lambda_context : dict
+        The lambda context
+
+    Returns
+    -------
+    dict
+        A dict of important properties for the request
+    """
+    if lambda_context is None:
```
Comment (mjtieman) [inline lifeomic_logging/logger.py]: Minor: use `if not lambda_context`, covers the None and empty dict cases.

File: lifeomic_logging/logger.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        return None
+
+    client_context = getattr(lambda_context, "client_context", None)
+    if not client_context:
+        return None
+
+    custom = getattr(client_context, "custom", {})
```
Comment (mjtieman) [inline lifeomic_logging/logger.py]: Minor: let's default to `None`, don't need to create a dict instance we won't use

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "python.pythonPath": ".venv/bin/python3.7"
```
Comment (simons5593) [inline .vscode/settings.json]: As this is ide settings, this is nbd but you probably want python3.6.

**Review by simons5593 — DISMISSED** [review]

File: lifeomic_logging/logger.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        "correlationId": request_correlation_id,
+        "requestId": request_id,
+        "user": request_user,
+    }
+
+
+def _logger(name: str, handler, level: int = INFO) -> Logger:
+    log = getLogger(name)
+
+    if not len(log.handlers):
```
Comment (mjtieman) [inline lifeomic_logging/logger.py]: Why get the length, can we just do a `if not log.handlers` check here?

File: lifeomic_logging/logger.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        textformatter = _JSONFormatter()
+        handler.setFormatter(textformatter)
+        log.addHandler(handler)
+
+    if not len(log.filters):
```
Comment (mjtieman) [inline lifeomic_logging/logger.py]: Same above

**Review by simons5593 — DISMISSED** [review]

File: .vscode/settings.json (hunk 4 above)
Comment (mschroering) [reply .vscode/settings.json]: I removed it and added to .gitignore.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,74 @@
...
+![Travis (.org) branch](https://img.shields.io/travis/lifeomic/logging-py/master.svg?style=for-the-badge)
+![PyPI status](https://img.shields.io/pypi/status/lifeomic_logging.svg?style=for-the-badge)
+![Downloads](https://img.shields.io/pypi/dw/lifeomic_logging?style=for-the-badge)
+![GitHub release](https://img.shields.io/github/release/lifeomic/logging-py.svg?style=for-the-badge)
+
+## Getting Started
+
+### Dependencies
+
+* [Python 3](https://www.python.org/download/releases/3.0/) version >= 3.7
```
Comment (mjtieman) [inline README.md]: 3.6

**Review by mjtieman — APPROVED** [review]
LGTM

**Review by simons5593 — APPROVED** [review]

**Pull request merged**